src/aux/MPIConvenience.cpp

```diff
@@ -0,0 +1,108 @@
+#include "MPIConvenience.h"
+#include <limits>
+#include <mpi.h>
+#include <vector>
+
+namespace {
+constexpr std::size_t DataIncrement = std::numeric_limits<int>::max();
+constexpr int Tag = 1000;
+
+std::vector<MPI_Request> largeIsend(const void* sendbuf, std::size_t sendsize, std::size_t senddisp,
+                                    MPI_Datatype sendtype, int dest, int tag, MPI_Comm comm) {
```

Comment:
Can you add a comment explaining in which way largeIsend differs from Isend and why it is needed? In the end, it is an extension of Isend to std::size_t?

Reply:
Yes; exactly. That's all it is, basically. We run multiple `Isend`s, if the data is larger than the integer limit.

Other methods to handle that would have been to use extra data types.

Reply:
However, the actual main reason all these extra methods are needed are the offsets—i.e. to have a scatter/alltoall that is _in total_ larger than the integer limit. Usually, we don't cross the integer boundary for a single send, i.e. the `largeIsend`/`largeIrecv` is more secondary.